core/DataAccess/LogAggregator.php

```diff
@@ -292,8 +293,17 @@ private function createTemporaryTable($unprefixedSegmentTableName, $segmentSelec
         } catch (\Exception $e) {
             if ($readerDb->isErrNo($e, \Piwik\Updater\Migration\Db::ERROR_CODE_TABLE_EXISTS)) {
                 return;
+            } elseif ($readerDb->isErrNo($e, \Piwik\Updater\Migration\Db::ERROR_CODE_REQUIRES_PRIMARY_KEY)
```

Comment:
haven't checked that, but which error is thrown earlier? The table exists or the primary key error? If it's the primary key error the second query might throw a table exists, which isn't ignored then...

Reply:
@sgiehl I checked the table exists error would be triggered first but adding it just to be safe 👍